src/cluster_legacy.c

```diff
@@ -2990,7 +2991,52 @@ int clusterProcessPacket(clusterLink *link) {
+                    if (master && !memcmp(master->shard_id, sender->shard_id, CLUSTER_NAMELEN)) {
+                        /* 'sender' was a primary and was in the same shard as `master`, its new primary */
+                        if (sender->configEpoch > senderConfigEpoch) {
+                            serverLog(LL_NOTICE,
+                                    "Ignore stale message from %.40s (%s) in shard %.40s;"
+                                    " gossip config epoch: %llu, current config epoch: %llu", 
+                                    sender->name,
+                                    sender->human_nodename,
+                                    sender->shard_id,
+                                    (unsigned long long)senderConfigEpoch,
+                                    (unsigned long long)sender->configEpoch);
+                        } else {
+                            /* A failover occurred in the shard where `sender` belongs to and `sender` is no longer
+                             * a primary. Update slot assignment to `master`, which is the new primary in the shard */
+                            int slots = clusterMoveNodeSlots(sender, master);
+                            serverLog(LL_NOTICE, "A failover occurred in shard %.40s; node %.40s (%s)"
+                                    " lost %d slot(s) to node %.40s (%s)",
+                                    sender->shard_id,
+                                    sender->name,
+                                    sender->human_nodename,
+                                    slots,
+                                    master->name,
+                                    master->human_nodename);
+                        }
+                    } else {
+                        /* `sender` was moved to another shard and has become a replica, remove its slot assignment */
+
+                        int slots = clusterDelNodeSlots(sender);
+                        if (master != NULL) {
+                            serverLog(LL_NOTICE, "Node %.40s (%s) moved from shard %.40s to shard %.40s"
+                                    " and became a slave; removed all %d slot(s) it used to own",
+                                    sender->name,
+                                    sender->human_nodename,
+                                    sender->shard_id,
+                                    master->shard_id,
+                                    slots);
+                        } else {
+                            /* We have no knowledge of `master` */
+                            serverLog(LL_NOTICE, "Node %.40s (%s) is no longer a master in shard %.40s;"
+                                    " removed all %d slot(s) it used to own",
+                                    sender->name,
+                                    sender->human_nodename,
+                                    sender->shard_id,
+                                    slots);
+                        }
```

Comment:
```suggestion
                            serverLog(LL_NOTICE, "Node %.40s (%s) is no longer master of shard %.40s; removed all %d slot(s) it used to own",
                                    sender->name,
                                    sender->human_nodename,
                                    sender->shard_id,
                                    slots);
                       if (master != NULL) {
                           serverLog(LL_NOTICE, "Node %.40s (%s) is now part of shard %.40s, ... whatever);
                        }
```
I don't necessarily want to reduce the amount of logging, but want to reduce the amount and permutations of logs.

Reply:
make sense. updating